collect_app/src/main/java/org/odk/collect/android/activities/FormEntryActivity.java

```diff
@@ -1617,27 +1611,7 @@ private void createConstraintToast(FormIndex index, int saveStatus) {
                 return;
         }
 
-        showCustomToast(constraintText, Toast.LENGTH_SHORT);
-    }
-
-    /**
-     * Creates a toast with the specified message.
-     */
-    private void showCustomToast(String message, int duration) {
```

Comment:
This method displayed a toast in the center of a screen. I didn't like it but you definitely shouldn't change it as a part of this pr. I understand you because I've always struggled with not implementing many not related changes too :smile: but could you leave only those needed? I'm' talking about Lambda Expressions as well.

Reply:
If we want lambda expression you can create a quick win issue since probably we can just replace all of them using Android Studio.

Reply:
The idea of removing this was to create two customized toast message theme which is controlled from within `ToastUtils`. Right now, we get a dark background with white text in both of the themes. Should a create a light/dark theme for this as well?

Reply:
Because in the dark theme, the screen is already dark and the Toast window is as visible as it is Light theme

Reply:
```
    <!-- Default Gravity setting for the system Toast view. Equivalent to: Gravity.CENTER_HORIZONTAL | Gravity.BOTTOM -->
    <integer name="config_toastDefaultGravity">0x00000051</integer>
```

This is the default gravity of the Toast

Reply:
Also, the textview's gravity is center_horizontal too by default

`transient_notification.xml`

Reply:
@lognaturel @grzesiek2010 So should I remove all the unrelated changes? (lamdas, code indentation, cleanup)

Reply:
especially lambdas I think because as I said if we want to use lambdas let's fix all classes.

Reply:
Ok, thanks for the details, @shobhitagarwal1612, that helps make it clear this is not an unrelated change. I saw you used a detailed message in some commits, this kind of information about why you're doing something can be really helpful to put there (e.g. "All toasts go through ToastUtils so theme can be controlled in one place.")

As far as whether it does make sense to theme the toasts, I think they're legible enough as is. I have not seen white toasts anywhere.

Reply:
When it comes to the background I didn't test your changes but for example here https://www.google.pl/search?q=android+dark+theme+toast&source=lnms&tbm=isch&sa=X&ved=0ahUKEwiPgZnh84raAhVCLFAKHV6RD_AQ_AUICigB&biw=1855&bih=917#imgrc=6ITl-8K6iwRFwM:

dark background and dark toast is well visible for me.

Reply:
It looks as if there is a white toast in Android 8 https://www.androidpolice.com/2017/10/26/android-8-1-feature-spotlight-toast-messages-now-white-better-match-rest-ui/

Reply:
Please correct me if I understood this wrong. So the plan is to revert back the lambdas but keep the Toast change, right? @lognaturel @grzesiek2010

Reply:
> It looks as if there is a white toast in Android 8 https://www.androidpolice.com/2017/10/26/android-8-1-feature-spotlight-toast-messages-now-white-better-match-rest-ui/

So this means the Toasts look different on different phones. Should we try to make it consistent for all or leave them as they are?

Reply:
As @lognaturel said commits are quite clear so leave it if you want, just remember for the future. As I said I know The Boy Scout Rule and I also try to improve the code but in this pr, you have some completely unrelated like in HierarchyElement that's why I pointed this out.

Reply:
> So this means the Toasts look different on different phones. Should we try to make it consistent for all or leave them as they are?

It's a default android behavior so we shouldn't change it.

Reply:
Great to know that there's a name for the rule too ` The Boy Scout Rule`
As mentioned @lognaturel,  @grzesiek2010 if the unrelated changes are looking fine to you as well then let's keep them as they are. I totally agree I did some completely unrelated changes which I shouldn't have and thanks for pointing it out to me. I need to be more careful while doing the refactors